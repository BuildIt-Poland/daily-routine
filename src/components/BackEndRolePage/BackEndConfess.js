import React from 'react';

import { CONFESS } from '../../constants/roleActions';
import { Robot } from '../Characters';

function BackEndConfess() {
  return <Robot pose={CONFESS} />;
}

export default BackEndConfess;
