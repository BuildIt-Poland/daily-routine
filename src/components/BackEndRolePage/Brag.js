import React from 'react';

import { CONFIDENT } from '../../constants/characterMoods';
import { Robot } from '../Characters';

function Brag() {
  return <Robot mood={CONFIDENT} />;
}

export default Brag;
