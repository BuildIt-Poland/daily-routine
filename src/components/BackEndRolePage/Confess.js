import React from 'react';

import { SAD } from '../../constants/characterMoods';
import { Robot } from '../Characters';

function Confess() {
  return <Robot mood={SAD} />;
}

export default Confess;
