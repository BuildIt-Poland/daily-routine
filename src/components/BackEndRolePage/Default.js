import React from 'react';

import { CONFUSED } from '../../constants/characterMoods';
import { Robot } from '../Characters';

function Default() {
  return <Robot mood={CONFUSED} />;
}

export default Default;
