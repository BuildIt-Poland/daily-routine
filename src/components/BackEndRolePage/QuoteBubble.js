import styled from 'styled-components';

import { spacingMedium } from '../../styles/designTokens';
import QuoteBubbleBase from '../QuoteBubble';

const QuoteBubble = styled(QuoteBubbleBase)`
  margin-bottom: -${spacingMedium};
`;

export default QuoteBubble;
