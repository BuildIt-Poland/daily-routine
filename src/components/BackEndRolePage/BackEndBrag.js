import React from 'react';
import sample from 'lodash.sample';

import { BRAG } from '../../constants/roleActions';
import { Robot } from '../Characters';
import QuoteBubble from '../QuoteBubble';

import QUOTES from '../../quotes/backEndQuotes';

// TODO Replace with auto-generated message
const QUOTE = sample(QUOTES);

function BackEndBrag() {
  return (
    <>
      <QuoteBubble quote={QUOTE} />
      <Robot pose={BRAG} />
    </>
  );
}

export default BackEndBrag;
