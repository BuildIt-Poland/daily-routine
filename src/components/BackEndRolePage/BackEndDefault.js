import React from 'react';

import { DEFAULT } from '../../constants/roleActions';
import { THOUGHT } from '../../constants/speechBubbleVariant';
import { Robot } from '../Characters';
import QuoteBubble from '../QuoteBubble';

function BackEndDefault() {
  return (
    <>
      <Robot pose={DEFAULT} />
      <QuoteBubble variant={THOUGHT} noCopyToClipboard />
    </>
  );
}

export default BackEndDefault;
