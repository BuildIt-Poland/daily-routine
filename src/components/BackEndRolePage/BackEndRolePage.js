import React from 'react';

import { BACK_END_ROLE, BACK_END_ROLE_LABEL } from '../../constants/roles';
import ErrorBoundary from '../ErrorBoundary';
import Page from '../Page';
import ActionButtons from '../ActionButtons';
import RoleBar from '../RoleBar';
import Section from '../Section';
import RoleContent from '../RoleContent';
import Character from '../Character';
import Robot from '../Characters/Robot';
import QuoteBubble from './QuoteBubble';

function BackEndRolePage() {
  return (
    <ErrorBoundary>
      <Page title={BACK_END_ROLE_LABEL}>
        <RoleBar label={BACK_END_ROLE_LABEL} />
        <Section>
          <RoleContent>
            <QuoteBubble />
            <Character>
              <Robot />
            </Character>
          </RoleContent>
        </Section>
        <ActionButtons role={BACK_END_ROLE} />
      </Page>
    </ErrorBoundary>
  );
}

export default BackEndRolePage;
