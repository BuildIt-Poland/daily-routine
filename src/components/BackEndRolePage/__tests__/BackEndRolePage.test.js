import React from 'react';
import { create } from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import BackEndRolePage from '../BackEndRolePage';

jest.mock('../../Characters', () => ({
  Robot: 'Robot'
}));
jest.mock('../../RoleActionButton', () => 'RoleActionButton');
jest.mock('../../RoleBar', () => 'RoleBar');
jest.mock('../../QuoteBubble', () => 'QuoteBubble');

describe('COMPONENT - BackEndRolePage', () => {
  it('renders correctly for /backend path', () => {
    const component = create(
      <MemoryRouter initialEntries={['/backend']}>
        <BackEndRolePage />
      </MemoryRouter>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly for /backend/brag path', () => {
    const component = create(
      <MemoryRouter initialEntries={['/backend/brag']}>
        <BackEndRolePage />
      </MemoryRouter>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly for /backend/confess path', () => {
    const component = create(
      <MemoryRouter initialEntries={['/backend/confess']}>
        <BackEndRolePage />
      </MemoryRouter>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
