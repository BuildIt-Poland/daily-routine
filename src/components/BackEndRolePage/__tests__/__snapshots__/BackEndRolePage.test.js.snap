// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - BackEndRolePage renders correctly for /backend path 1`] = `
Array [
  <RoleBar
    label="Back End Developer"
  />,
  .c0 {
  position: relative;
  padding: 2rem;
}

<section
    className="c0"
  >
    <Robot
      gender="male"
      pose="default"
    />
    <QuoteBubble
      noCopyToClipboard={true}
      variant="thought"
    />
    <ActionButtons
      role="backend"
    />
  </section>,
]
`;

exports[`COMPONENT - BackEndRolePage renders correctly for /backend/brag path 1`] = `
Array [
  <RoleBar
    label="Back End Developer"
  />,
  .c0 {
  position: relative;
  padding: 2rem;
}

<section
    className="c0"
  >
    <Robot
      gender="male"
      pose="brag"
    />
    <QuoteBubble
      quote="Yesterday I was implementing PUT method for missing RESTful API endpoint."
    />
    <ActionButtons
      role="backend"
    />
  </section>,
]
`;

exports[`COMPONENT - BackEndRolePage renders correctly for /backend/confess path 1`] = `
Array [
  <RoleBar
    label="Back End Developer"
  />,
  .c0 {
  position: relative;
  padding: 2rem;
}

<section
    className="c0"
  >
    <Robot
      gender="male"
      pose="confess"
    />
    <QuoteBubble
      quote="Yesterday I was implementing PUT method for missing RESTful API endpoint."
    />
    <ActionButtons
      role="backend"
    />
  </section>,
]
`;
